README.md

```diff
@@ -1,206 +1,193 @@
+
+const getCarIdProp = createIdSelector(props => props.carId);
+const getRunningCar = createSelector(
+  [
+    getRunningCars,
+    getCarIdProp
+  ],
+  (cars, carId) => cars.find(car => car.id === carId)
+);
+const getRunningCarPassengers = createSelector(
+  [
+    getRunningCar,
+    getPassengers
+  ],
+  (car, passengers) => car.passengerIds.map(id => passengers[id])
+);
+
+const passengersCar1_0 = getRunningCarPassengers(state, { carId: 1 }); // computes
+const passengersCar2 = getRunningCarPassengers(state, { carId: 2 }); // computes
+const passengersCar1_1 = getRunningCarPassengers(state, { carId: 1 }); // cached
+
+assert(passengersCar1_0 === passengersCar1_1);
+assert(getRunningCars.recomputations() === 2);
 ```
 
-Again, the same number of computations, because nothing relevant has changed, awesome!
+This way, `redux-views` knows that `getRunningCarPassengers` will probably give different results for each `carId`, effectively allowing it to memoize the value for each one.
 
-Now let's try making a relevant change, the loading status of user "2":
+It's good to know that, by default, `redux-views` will keep those memoized values for as long as the application runs. However, it provides two ways of clearing the cache, either manually or with a ref count which automatically clears the cache when the selector is not used anymore. This is something internal and rarely used for individual projects, as it's meant for bindings with state management libraries.
 
-```js
-const newState = {
-  ...state,
-  loadingUsers: { ...state.loadingUsers, '2': true }
-}
+## Migrating from reselect@4.0
 
-getUserInfo(newState, {id: '1'})
-getUserInfo(newState, {id: '2'})
+This version of `redux-views` removes the variadic overload of `createSelector` from `reselect@4.0`, which means that all calls that were using this overload will need to be changed. For example:
 
-getUserInfo.recomputations() // => 3
-getIsUserLoading.recomputations() // => 3
-getUser.recomputations() // => 2
+```js
+createSelector(
+  getRunningCars,
+  getPassengers,
+  (cars, passengers) => ...
+);
 ```
 
-Notice how `getUserInfo` and `getIsUserLoading` have increased the number of recomputations, cool! However, the recomputations of `getUser` remain the same, right? That's because the part of the state that's relevant to `getUser` has not changed.
-
-I know what you must be thinking: what happens when a selector depends on more than one different id-selector?
-
-I'm glad that you asked. Let's try it:
+becomes
 
 ```js
-const getUsers = state => state.users
+createSelector(
+  [
+    getRunningCars,
+    getPassengers,
+  ],
+  (cars, passengers) => ...
+);
+```
 
-const getPropIdA = createIdSelector(({idA}) => idA)
-const getPropIdB = createIdSelector(({idB}) => idB)
+That's the only breaking change for the simple usage. `redux-views` passes all the tests from `reselect@4.0`. There's a codemod in the works that will allow to replace all of these automatically.
```

Comment:
I think I lied here... Now I see it's not the only breaking change, there's also `defaultMemoize` and `createSelectorCreator` that were removed. We'll have to document how to migrate those as well....

Reply:
> We'll have to document how to migrate those as well....

I don't know about that... I'm of the opinion that those were implementation details and I would much rather to ask the community how and why they were using those. I hope that in most cases we will be able to either suggest another alternative or that we will find a use-case to improve the API of this library.

In short: I wouldn't worry too much about that.